sdk/tests/iot/provisioning/test_az_iot_provisioning_client_parser.c

```diff
@@ -682,6 +848,12 @@ int test_az_iot_provisioning_client_parser()
     cmocka_unit_test(test_az_iot_provisioning_client_operation_complete_translate_succeed),
     cmocka_unit_test(test_az_iot_provisioning_client_logging_succeed),
     cmocka_unit_test(test_az_iot_provisioning_client_no_logging_succeed),
+    cmocka_unit_test(
+        test_az_iot_provisioning_client_parse_received_topic_and_payload_json_custom_payload_succeed),
+    cmocka_unit_test(
+        test_az_iot_provisioning_client_parse_received_topic_and_payload_json_custom_payload_empty_succeed),
+    cmocka_unit_test(
+        test_az_iot_provisioning_client_parse_received_topic_and_payload_json_custom_payload_null_succeed),
```

Comment:
Are you able to double check that we have a test for when the options are `NULL` and when they're not `NULL`?

Reply:
Options can't be NULL:

> Roll-back from feature branch: member function options parameter cannot be NULL. (https://github.com/Azure/azure-sdk-for-c/pull/2188#discussion_r860141070)